hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieSparkCompactor.java

```diff
@@ -43,19 +48,25 @@ public HoodieSparkCompactor(BaseHoodieWriteClient<T, JavaRDD<HoodieRecord<T>>, J
   }
 
   @Override
-  public void compact(HoodieInstant instant) throws IOException {
+  public void compact(HoodieInstant instant) {
     LOG.info("Compactor executing compaction " + instant);
     SparkRDDWriteClient<T> writeClient = (SparkRDDWriteClient<T>) compactionClient;
-    JavaRDD<WriteStatus> res = writeClient.compact(instant.getTimestamp());
+    HoodieWriteMetadata<JavaRDD<WriteStatus>> compactionMetadata = writeClient.compact(instant.getTimestamp());
+    List<HoodieWriteStat> writeStats = compactionMetadata.getCommitMetadata().get()
+        .getPartitionToWriteStats()
+        .values()
+        .stream()
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
     this.context.setJobStatus(this.getClass().getSimpleName(), "Collect compaction write status");
```

Comment:
now with this patch we no need to set JobStatus here, since no spark job will trigger, pls delete this line

Reply:
Changed. Thanks a lot for your review.